astropy/cosmology/funcs/comparison.py

```diff
@@ -0,0 +1,439 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Comparison functions for `astropy.cosmology.Cosmology`.
+
+This module is **NOT** public API. To use these functions, import them from
+the top-level namespace -- :mod:`astropy.cosmology`.
+This module will be moved.
+"""
+
+from __future__ import annotations
+
+import functools
+import inspect
+from typing import Any, Callable, Tuple, Union
+
+import numpy as np
+from numpy import False_, True_, ndarray
+
+from astropy import table
+from astropy.cosmology.core import Cosmology
+
+__all__ = []  # Nothing is scoped here
+
+
+##############################################################################
+# PARAMETERS
+
+_FormatType = Union[bool, None, str]
+_FormatsT = Union[_FormatType, Tuple[_FormatType, ...]]
+_CompFnT = Callable[[Any, _FormatType], Cosmology]
+
+_COSMO_AOK = (None, True_, False_, "astropy.cosmology")
```

Comment:
Would it be a good idea to use a `set` instead of a `tuple`?

Reply:
Yes, it would.

Reply:
✅